### DISCUSSION TITLE
What causes the graphics to break?

### DISCUSSION THREAD

**Discussion by LISPos** [unoplatform/uno#9581] [Q&A]
I ran it in skia.gtk environment
No borders for buttons or controls are drawn.
In wasm and wpf.host , the output is normal.
A problem occurs in skia.gtk.
Is the graphics output different on linux platforms?

![b1](https://user-images.githubusercontent.com/20689998/186295365-3e521f1a-9cf8-4ee8-91d0-b52b26803f4c.PNG)

**Comment by jeromelaban** [comment]
Thanks for the report. Which version of Uno are you using?

**Reply by LISPos** [reply]
Thanks for your reply.
I am using version (Uno 4.4.13.8  / Ubuntu 20.04  / VisualStudio2019).

**Reply by LISPos** [reply]
I found that the graphics were broken when I updated the version from 4.1.9 to 4.2.6.
Uno.UI
Uno.UI.Adapter.Microsoft.Extensions.Logging
Uno.UI.RemoteControl
Uno.UI.Skia.Gtk

**Reply by jeromelaban** [reply]
Thanks for the update. Can you try with the uno dev (prerelease) packages to see if this helps?

**Reply by LISPos** [reply]
I checked.
[4.1.8], [4.1.9], [4.3.5], [4.3.6] → No problem
[4.2.6], [4.3.8], [4.4.5], [4.4.13], [4.5.0-dev.598] → problem

**Reply by jeromelaban** [reply]
Thanks. This looks related to the use of OpenGL, we'll investigate.

In the meantime, can you try with the following to use software rendering:

```csharp
var host = new GtkHost(() => new App());
host.RenderSurfaceType = RenderSurfaceType.Software;
```

**Reply by jeromelaban** [reply]
Let's track in https://github.com/unoplatform/uno/issues/9605

**Reply by LISPos** [reply]
Characters are out of the Textbox area.
I'm testing it on Skia.Gtk.
I tested the latest versions of Uno as well as [4.1.8] [4.1.9] [4.3.5] [4.4.13].
Controls are unstable in Skia.Gtk.
Is this happening only to me?
I am looking to develop an application in a Linux environment.
Is it too early to use the Uno platform?
Thank you for reading my poor English.

![multi](https://user-images.githubusercontent.com/20689998/187361459-0a4dec50-bad9-49e8-b503-eb3ea3aab72c.PNG)

**Reply by jeromelaban** [reply]
Multiline `TextBox` is not yet fully supported on Skia.Gtk at this time. You can follow https://github.com/unoplatform/uno/issues/4309 for more details.

Can you tell which other controls are unstable on Skia.Gtk ?
